packages/functions/src/resample.ts

```diff
@@ -46,14 +47,40 @@ export function resample(_options: ResampleOptions = RESAMPLE_DEFAULTS): Transfo
 			}
 
 			for (const sampler of animation.listSamplers()) {
-				if (samplerTargetPaths.get(sampler) === 'weights') {
+				const targetPath = samplerTargetPaths.get(sampler)!;
+				if (targetPath === 'weights') {
```

Comment:
TODO: It's unfortunate that morph target weights are still skipped here, and it'd be nice to figure out the cause and fix that. See #290.

Reply:
In [spec](https://github.com/KhronosGroup/glTF/blob/d202b79a816943beacdd9be6c936b63a1a9c0ea5/specification/2.0/Specification.adoc?plain=1#L2555), animation samplers targeted at weights of morph targets has `SCALAR` accessor type, and the [current resample function](https://github.com/donmccurdy/glTF-Transform/blob/eae7adab2d221daa98e5d9cd6061ea1015d2b9b2/packages/functions/src/resample.ts#L106) takes this as the [element size](https://github.com/donmccurdy/glTF-Transform/blob/eae7adab2d221daa98e5d9cd6061ea1015d2b9b2/packages/core/src/properties/accessor.ts#L169) by [`getElement`](https://github.com/donmccurdy/glTF-Transform/blob/eae7adab2d221daa98e5d9cd6061ea1015d2b9b2/packages/core/src/properties/accessor.ts#L402), but it is [defined](https://github.com/KhronosGroup/glTF/blob/d202b79a816943beacdd9be6c936b63a1a9c0ea5/specification/2.0/Specification.adoc?plain=1#LL2581C1-L2581C138) as

> A morph target animation frame is defined by a sequence of scalars of length equal to the number of targets in the animated morph target.

So morph target animation with more than 1 targets would suffer from the issue referenced here.

Since the [new impl](https://github.com/donmccurdy/keyframe-resample-wasm/blob/d6896fa73af70370abf711486cdcc9c0b9b861ba/assembly/resample.ts#L23) does not use element size from accessor, the linked issue might be fixed by this.

Reply:
Oh that's it, thank you! Found one other issue in the WASM implementation testing this, but it does look like this PR will now fix #290. 🎊